src/ophyd_async/core/signal_backend.py

```diff
@@ -27,7 +27,7 @@ async def put(self, value: Optional[T], wait=True, timeout=None):
         """Put a value to the PV, if wait then wait for completion for up to timeout"""
 
     @abstractmethod
-    async def get_descriptor(self, source: str) -> Descriptor:
+    async def get_datakey(self, source: str) -> DataKey:
```

Comment:
There is a longstanding issue that tables are not representable under the current scheme as we can only put a single entry in the descriptor, where we would like to put one entry for each column in the table. To fix this we would eventually like to make this:

```suggestion
    async def get_datakey(self, name: str, source: str) -> Dict[str, DataKey]:
```

...